src/bin/elfshaker/pack.rs

```diff
@@ -33,6 +36,11 @@ pub(crate) fn run(matches: &ArgMatches) -> Result<(), Box<dyn Error>> {
         })
         .transpose()?;
 
+    if snapshots_from.is_some() && snapshots0_from.is_some() {
+            error!("Cannot specify both --snapshots-from and --snapshots0-from!");
+            return Err("Invalid options!".into());
+        }
```

Comment:
Rather than expressing this a separate test-and-then-compute, could you do it with a single pattern match?

There is also a third thing to consider, if wanting to give the user good errors: remaining arguments are interpreted as snapshots via `indexes`, which if I'm reading right get superseded by snapshots, but maybe we should error on that to avoid confusing mistakes.

Something like:

```rust
let open = |filename| if filename == "-" { Box::new(stdin()) } else { Box::new(File::open(filename)) };
let snapshots = match (snapshots_from, snapshots0_from, indexes[..]) {
  (Some(s), None, []) => packs_from_list(open(file), delim),
  (None, Some(s), []) => packs_from_list(open(file), delim),
  (None, None, _) => indexes,
  _ => can't specify more than one!
};
```

If requested, I can be more specific through experimentation.

Reply:
That's a fair point, let me make the change :)